contracts/simple-ica-controller/src/ibc.rs

```diff
@@ -111,11 +117,60 @@ pub fn ibc_packet_ack(
 #[allow(clippy::unnecessary_wraps)]
 fn acknowledge_dispatch(
     _deps: DepsMut,
+    _env: Env,
     _caller: String,
-    _ack: StdAck,
+    callback: Option<String>,
+    msg: IbcPacketAckMsg,
 ) -> Result<IbcBasicResponse, ContractError> {
     // TODO: actually handle success/error?
-    Ok(IbcBasicResponse::new().add_attribute("action", "acknowledge_dispatch"))
+    match callback {
+        Some(callback) => {
+            // Send IBC packet ack message to another contract
+            let msg = WasmMsg::Execute {
+                contract_addr: callback.clone(),
+                msg: to_binary(&ReceiveIbcResponseMsg { msg })?,
+                funds: vec![],
+            };
+            Ok(IbcBasicResponse::new()
+                .add_attribute("action", "acknowledge_dispatch")
+                .add_attribute("callback_address", callback)
+                .add_message(msg))
+        }
+        None => Ok(IbcBasicResponse::new().add_attribute("action", "acknowledge_dispatch")),
+    }
+}
+
+fn acknowledge_query(
+    deps: DepsMut,
+    env: Env,
+    caller: String,
+    callback: Option<String>,
+    msg: IbcPacketAckMsg,
+) -> Result<IbcBasicResponse, ContractError> {
+    // store IBC response for later querying from the smart contract??
+    LATEST_QUERIES.save(
```

Comment:
Not sure how useful it is, but since it is indexed by the caller, why not?
We can revisit later